contracts/modules/TokenExchanger.sol

```diff
@@ -33,139 +35,238 @@ contract TokenExchanger is OnlyOwnerModule {
-    // The address of the contract collecting fees for Argent.
-    address public feeCollector;
-    // The Argent fee in 1-per-10000.
-    uint256 public feeRatio;
+    // The address of the Paraswap Proxy contract
+    address public paraswapProxy;
+    // The address of the Paraswap contract
+    address public paraswapSwapper;
+    // The label of the referrer
+    string public referrer;
+    // Authorised exchanges
+    mapping(address => bool) public authorisedExchanges;
 
     event TokenExchanged(address indexed wallet, address srcToken, uint srcAmount, address destToken, uint destAmount);
 
+
+    // *************** Constructor ********************** //
+
     constructor(
         IModuleRegistry _registry,
         IGuardianStorage _guardianStorage,
-        address _kyber,
-        address _feeCollector,
-        uint _feeRatio
+        address _paraswap,
+        string memory _referrer,
+        address[] memory _authorisedExchanges
     )
         BaseModule(_registry, _guardianStorage, NAME)
         public
     {
-        kyber = _kyber;
-        feeCollector = _feeCollector;
-        feeRatio = _feeRatio;
+        paraswapSwapper = _paraswap;
+        paraswapProxy = IAugustusSwapper(_paraswap).getTokenTransferProxy();
+        referrer = _referrer;
+
+        for (uint i = 0; i < _authorisedExchanges.length; i++) {
+            authorisedExchanges[_authorisedExchanges[i]] = true;
+        }
     }
 
     /**
-     * @dev Lets the owner of the wallet execute a trade.
+     * @dev Lets the owner of the wallet execute a "sell" trade (fixed source amount, variable destination amount).
      * @param _wallet The target wallet
      * @param _srcToken The address of the source token.
-     * @param _srcAmount The amoutn of source token to trade.
      * @param _destToken The address of the destination token.
-     * @param _maxDestAmount The maximum amount of destination token accepted for the trade.
-     * @param _minConversionRate The minimum accepted rate for the trade.
-     * @return The amount of destination tokens that have been received.
+     * @param _srcAmount The exact amount of source tokens to sell.
+     * @param _minDestAmount The minimum amount of destination tokens required for the trade.
+     * @param _expectedDestAmount The expected amount of destination tokens (used only in ParaSwap's events).
+     * @param _path Sequence of sets of weighted ParaSwap routes. Each route specifies an exchange to use to convert a given (exact) amount of a given
+     * source token into a given (minimum) amount of a given destination token. The path is a sequence of sets of weighted routes where the destination
+     * token of a set of weighted routes matches the source token of the next set of weighted routes in the path.
+     * @param _mintPrice gasPrice (in wei) at the time the gas tokens were minte by ParaSwap. 0 means gas token will not be used by ParaSwap
      */
-    function trade(
+    function multiSwap(
```

Comment:
Why not name this method `sell` since the other one is called `buy`? It will probably avoid some confusion.

Reply:
Yes, I hesitated but in the end decided to use naming consistent with Paraswap's contract (which uses `buy` and `multiSwap`). Open to both.

Reply:
I would go for `buy` and `sell` irrespective of their names in Paraswap. In the end it will make it clearer for us what each method does (i.e. simpler for clients, support, etc).